src/SIL.Machine.AspNetCore/Services/ClearMLService.cs

```diff
@@ -204,16 +216,42 @@ public async Task<bool> StopTaskAsync(string id, CancellationToken cancellationT
         CancellationToken cancellationToken = default
     )
     {
+        await AuthorizeAsync();
         var request = new HttpRequestMessage(HttpMethod.Post, $"{_options.CurrentValue.ApiServer}/{service}.{action}")
         {
             Content = new StringContent(body.ToJsonString(), Encoding.UTF8, "application/json")
         };
+        request.Headers.Add("Authorization", $"Bearer {authorizationToken}");
+        HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
+        string result = await response.Content.ReadAsStringAsync();
+        return (JsonObject?)JsonNode.Parse(result);
+    }
+
+    private async Task AuthorizeAsync(CancellationToken cancellationToken = default)
```

Comment:
Does this access token expire? If so, we will need to check if it is expired and request a new access token. Also, this makes the service stateful. It is possible for multiple threads to call this method concurrently causing race conditions. Locking should be used to avoid these issues.

Reply:
Could we change ClearMLService to have a Scoped rather than Singleton lifetime?

Reply:
Check if creating a new access token will invalidate existing ones.  Check if they have lifetimes.

Reply:
Singleton should be acceptable as long as the access token lifetime and race conditions are taken into account.

Reply:
Changed from getting a new auth token each time a request is made to periodically refreshing the authentication token (every hour).  That removes any race condition, accounts for the lifetime of the token (max 30 days - it would have broken after a month and we would not know why - good catch) and should be ok if the token is being refreshed when there is a call, as the "old" tokens are not invalidated until their timer runs out.